core/arch/arm/mm/core_mmu.c

```diff
@@ -85,7 +85,7 @@ static struct memaccess_area nsec_shared[] = {
 	MEMACCESS_AREA(CFG_SHMEM_START, CFG_SHMEM_SIZE),
 };
 
-#ifdef CFG_TEE_SDP_MEM_BASE
+#ifdef CFG_SECURE_DATA_PATH
```

Comment:
Platforms may decide to enable SDP but define locally in their plat-xxx/main.c (for example) the SDP memory areas, not relying on the generic single SDP memory area defined by `CFG_TEE_SDP_MEM_BASE/_SIZE`. 

Maybe a solution is to remove this generic `register_sdp_mem(CFG_TEE_SDP_MEM_BASE, CFG_TEE_SDP_MEM_SIZE);` from core_mmu.c and let each platform define their area(s) based on their private platform config info.

Reply:
@etienne-lms yeah that may be cleaner but for the moment all platforms have only one area so it is convenient to keep `register_sdp_mem()` in `core_mmu.c`...

So back to my proposal, how about doing just the following in this PR? (no more no less). Note the `#else` part that is currently missing in the proposed update. Wouldn't this fix the [initial problem](https://github.com/OP-TEE/optee_os/pull/2014#issuecomment-352051399) ?:
```diff
diff --git a/core/arch/arm/include/mm/core_mmu.h b/core/arch/arm/include/mm/core_mmu.h
index fe033508..9fc51439 100644
--- a/core/arch/arm/include/mm/core_mmu.h
+++ b/core/arch/arm/include/mm/core_mmu.h
@@ -218,9 +218,13 @@ struct core_mmu_phys_mem {
                __register_memory1_ul(#addr, (type), (addr), (size), \
                                   phys_mem_map_section, __COUNTER__)
 
+#ifdef CFG_SECURE_DATA_PATH
 #define register_sdp_mem(addr, size) \
                __register_memory1(#addr, MEM_AREA_SDP_MEM, (addr), (size), \
                                   phys_sdp_mem_section, __COUNTER__)
+#else
+#define register_sdp_mem(addr, size)
+#endif
 
 #define register_nsec_ddr(addr, size) \
                __register_memory1(#addr, MEM_AREA_RAM_NSEC, (addr), (size), \
```

Reply:
Yes, this looks good to me: disabling `register_sdp_mem()` if `CFG_SECURE_DATA_PATH` if is disabled. This would leave platform defining their memory layout.

Reply:
Ok. I'll add the change. Thank you both!